src/Microsoft.Framework.DesignTimeHost/PluginHandler.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Framework.DesignTimeHost.Models.IncomingMessages;
+using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Common.DependencyInjection;
+
+namespace Microsoft.Framework.DesignTimeHost
+{
+    public class PluginHandler
+    {
+        private readonly Action<object> _sendMessageMethod;
+        private readonly IServiceProvider _hostServices;
+        private readonly IDictionary<string, IPlugin> _plugins;
+
+        public PluginHandler(IServiceProvider hostServices, Action<object> sendMessageMethod)
+        {
+            _sendMessageMethod = sendMessageMethod;
+            _hostServices = hostServices;
+            _plugins = new ConcurrentDictionary<string, IPlugin>(StringComparer.Ordinal);
```

Comment:
not sure about the half measure.  creating a `ConcurrentDictionary` and using it as an `IDictionary` won't prevent some issues e.g. registration races (need `GetOrAdd()` for that).  let's talk tomorrow about whether to stick with a regular `Dictionary` or use full breadth of protections `ConcurrentDictionary` can provide.